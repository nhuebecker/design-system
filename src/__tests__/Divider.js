import React from 'react'
import renderer from 'react-test-renderer'
import { Divider, theme } from '..'

describe('Divider', () => {
  test('renders', () => {
    const json = renderer.create(<Divider />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('color prop sets background-color', () => {
    const json = renderer.create(<Divider color="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
  })

  test('borderColor prop sets background-color', () => {
    const json = renderer.create(<Divider borderColor="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
  })

  test('bg prop sets background-color', () => {
    const json = renderer.create(<Divider bg="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
  })

  test('height prop', () => {
    const json = renderer.create(<Divider height="4px" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('height', '4px')
  })
})
