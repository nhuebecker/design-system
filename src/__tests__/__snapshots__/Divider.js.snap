// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider bg prop sets background-color 1`] = `
.c1 {
  margin-left: 0px;
  margin-right: 0px;
  background-color: #7280ed;
}

.c0 {
  height: 3px;
  border: 0;
  background-color: #7280ed;
}

<hr
  className="c0 c1"
  height="3px"
/>
`;

exports[`Divider borderColor prop sets background-color 1`] = `
.c1 {
  margin-left: 0px;
  margin-right: 0px;
}

.c0 {
  height: 3px;
  border: 0;
  background-color: #7280ed;
}

<hr
  className="c0 c1"
  height="3px"
/>
`;

exports[`Divider color prop sets background-color 1`] = `
.c1 {
  margin-left: 0px;
  margin-right: 0px;
  color: #7280ed;
}

.c0 {
  height: 3px;
  border: 0;
  background-color: #7280ed;
}

<hr
  className="c0 c1"
  color="accent"
  height="3px"
/>
`;

exports[`Divider height prop 1`] = `
.c1 {
  margin-left: 0px;
  margin-right: 0px;
}

.c0 {
  height: 4px;
  border: 0;
  background-color: #dde1e4;
}

<hr
  className="c0 c1"
  height="4px"
/>
`;

exports[`Divider renders 1`] = `
.c1 {
  margin-left: 0px;
  margin-right: 0px;
}

.c0 {
  height: 3px;
  border: 0;
  background-color: #dde1e4;
}

<hr
  className="c0 c1"
  height="3px"
/>
`;
